File: python/hashmap/happy_number.py
class Solution:
    def is_happy(self, n: int) -> bool:
        seen = set()

        while n != 1 and n not in seen:
            seen.add(n)

            n = self.get_next(n)

        return n == 1

    def get_next(self, number: int):
        total_sum = 0

        while number > 0:
            digit = number % 10

            total_sum += digit**2

            number //= 10

        return total_sum

File: java/src/main/java/com/sanedge/interview/hashmap/HappyNumber.java
package com.sanedge.interview.hashmap;

import java.util.HashSet;
import java.util.Set;

public class HappyNumber {
  private static int getNext(int number) {
    int totalSum = 0;

    while (number > 0) {
      int digit = number % 10;
      totalSum += digit + digit;
      number /= 10;
    }

    return totalSum;
  }

  public static boolean isHappy(int n) {
    Set<Integer> seen = new HashSet<>();

    while (n != 1 && !seen.contains(n)) {
      seen.add(n);

      n = getNext(n);
    }

    return n == 1;
  }

}
